### ISSUE TITLE
Giada ignores ch play/stop and plays back midi if transport rolls

### ISSUE THREAD

**Issue by CarloCattano** [monocasual/giada#499]
- OS: PatchboxOS - Raspberry pi
- Giada version: Giada 0.18.2 -Build date: Aug 17 2021

Giada ignores ch actions state. Playhead plays notes if present in actions when transport rolls , no matter if its in stop state. 

1. open Giada with a midi output to an audio software that receives the midi 
2. Add midi channel 
3. click on edit actions with the channel stoped 
4. Start Giada's global transport play
5. Look at the actions editor , the playhead is rolling and if you paint notes , they are played . No matter the state of mute / play / stop 

expected behaviour :
 If play -> Play(output) notes  ,  if stopped -> dont  , if muted -> dont , if solo-ed -> output notes 

**Screenshots**
![image](https://user-images.githubusercontent.com/17380530/129849655-ea369c32-dc87-4c0d-b6cd-4eaeeeecb8f2.png)

**Additional context**
No matter if one does click on stop , play or other combo . The clip wont stop playing back to the midi output .

Using raspberry pi 3, Carla for audio synthesis , giada output to Alsa Midi-through port .
The version I installed previously from kxstudio repos ( I think ) its working well without this issue

**Comment by CarloCattano** [comment]
Present on Windows too . 
Giada 0.18.1
Build date: Jul 24 2021

Playhead is not displayed , maybe was a more recent adition , but this behaviout is still present :-(

![image](https://user-images.githubusercontent.com/17380530/129850925-8516d0c0-9465-4a89-9e61-cf0634f45080.png)

**Comment by gvnnz** [comment]
Could this be related to #497 ?

**Comment by CarloCattano** [comment]
It happens in  0.18.2 on my build for the pi , and on  0.18.1 on windows (official build release)  , but on 0.15.2 is behaving as expected  ( raspbian repo )

**Comment by CarloCattano** [comment]
```
for (const Action& a : as)
		if (a.channelId == ch.id)
			if (ch.isPlaying() && !ch.isMuted() )//|| ch.isArmed())
				send_(ch, a.event);

}
```
in L46 of midiSender.cpp this seems to take care of it ?

**Comment by CarloCattano** [comment]
```c++
if (e.type == eventDispatcher::EventType::KEY_KILL ||
	    e.type == eventDispatcher::EventType::SEQUENCER_STOP ||
		e.type == eventDispatcher::EventType::CHANNEL_MUTE)
		send_(ch, MidiEvent(G_MIDI_ALL_NOTES_OFF));
}
´´´
Added this to midiSender.cpp L71 in order to generate note offs when the channel goes suddenly mute . Otherwise I was getting hanging notes

**Comment by CarloCattano** [comment]
> Could this be related to #497 ? Does it happen also with the latest release 0.18.1?

Sorry , I checked and meant that MIDI output is being played , ignoring clip state. But the midi data going to plugins is fine 
Then I guess this is safe to write to prevent that
> ```c++
> for (const Action& a : as)
>   if (a.channelId == ch.id)
>      if (ch.isPlaying() && !ch.isMuted() )//|| ch.isArmed())
>          send_(ch, a.event);
> }
> ```
> 
> in L46 of midiSender.cpp this seems to take care of it ?
> relates to #497 too , and works , but I dont know what consequences can develop

**Comment by Xpktro** [comment]
I'm experimenting this as well in the 0.18.2 version (windows). Would try on my pi build but I gues it will behave the same.

**Comment by CarloCattano** [comment]
It is @Xpktro, you can add my dirty fixes above, or take a look at my testing forks for now.i can't compile from last pushes, I'll wait for the good news ❤️

**Comment by gvnnz** [comment]
Should have been fixed in e2e9ae3cbaf4208bf0a02212e8e390efd600f5dd .

**Comment by CarloCattano** [comment]
Indeed ! thanks !
